docs/architecture/tuvali.md

```diff
@@ -0,0 +1,88 @@
+
+Since the advertisement from Peripheral is of connectable type, Central would send out a SCAN_REQ on receiving advertisement and gets the remaining 27 bytes of peripheral public key. 
+Post that, Central would derive a public key pair and send the its 32 bytes public key on Identify characteristic of Peripheral.
+
+Once the public keys are exchanged between Central and Peripheral, a set of keys are derived on both sides which would be used for encryption and decryption of data on the wire.
+
+> Note: Details about the the algorithm used for public key pair generation, key derivation is available in the spec.
+
+
+### 2. Data transfer
+Steps 7 to 11 mentioned in the above diagram explains how second stage is achieved. 
+Before VC is transferred, Central performs encryption and compression and communicate the resultant data size by writing to Response Size characteristic to Peripheral. The actual data transfer would happen on `Submit Response` characteristic.
+
+Since the maximum allowed write value for a characteristic is limited to 512 bytes. The actual VC data is split by a component called Chunker into multiple smaller chunks. After the split, the data is transferred on the `Submit response`characteritics one after another until all chunks are completely sent.
+
+Peripheral on the other hand, receives data on the `Submit response` characteristic. The received chunks are collected and final VC is assembled by a component called Assembler.
+
+At the end of sending one frame of data from Central. It would request for a transfer report via `Transfer report request` characteristic. Peripheral responds with a summary of missing/corrupted chunks sequence numbers via another `Transfer report summary` characteristic. 
+
+Central would read the Transfer report sumamry to understand if the Peripheral received all the chunks properly. If summary report has atleast one chunk sequence number. Central would send those specific chunks to Peripheral which is called Failure frame.
+
+The failure frame will be sent from Central repeatedly until Transfer report summary is successful. If during the process, Central reached the maximum allowed failure frame retry limit, the tranfer is halted, devices will be disconnected and an error is generated (Please refer to API documentation on how this error can be read).
+
+### 3. Connection closure
+#### Disconnect initiated by Peripheral:
+- On a sucessful data transfer 
+- On non recoverable error occured on Peripheral
+
+Peripheral notifies Central to perform disconnection via `Disconnect` characteristic mentioned in Step 12 of the above diagram.
+
+#### Disconnect initiated by Central:
+Central also performs disconnect in the following scenarios
+- On a succesful data transfer
+- Non recoverable error on Central
+- Peripheral is out of range/disconnected
+- Destroy Connection API
+
+As part of Connection closure. Both Central and Peripheral objects are cleaned up along with Wallet and Verifier objects.
```

Comment:
We can simplify the above statement as follows

`As the part of connection closure, both central and peripheral cleans the held resources, cryptographic keys, bluetooth resources, to ensure that the subsequent transfer happens smoothly.`

Reply:
Updated as per recommendation